projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -463,16 +463,15 @@ export class IgxFilteringService implements OnDestroy {
             for (let i = 0; i < expressionsTree.filteringOperands.length; i++) {
                 this.generateExpressionsList(expressionsTree.filteringOperands[i], expressionsTree.operator, expressionsUIs);
             }
+            expressionsUIs[expressionsUIs.length - 1].afterOperator = operator;
```

Comment:
I recall that beforeOperator of the first item and afterOperator of the last item have to null but I don't remember the exact reason. This line here could lead to a regression in the filtering logic. Please double check if you really need this code here and if so carefully test all filtering scenarios to avoid regressions.